Methods returning TemplateInstance reported as not blocking

### Describe the bug

When adding @RunOnVirtualThread on a method returning `TemplateInstance` the application does not build, with exception:
```
java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.resteasy.reactive.server.deployment.ResteasyReactiveProcessor#setupEndpoints threw an exception: java.lang.RuntimeException: java.lang.RuntimeException: Failed to process method 'no.nte.services.ucmeteringvalues.rest.AdminResource#cashflowevents'
        at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createEndpoints(EndpointIndexer.java:325)
        at io.quarkus.resteasy.reactive.server.deployment.ResteasyReactiveProcessor.setupEndpoints(ResteasyReactiveProcessor.java:656)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:858)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
        at java.base/java.lang.Thread.run(Thread.java:1583)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.lang.RuntimeException: Failed to process method 'org.acme.SomePage#get'
        at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createResourceMethod(EndpointIndexer.java:789)
        at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createEndpoints(EndpointIndexer.java:414)
        at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createEndpoints(EndpointIndexer.java:292)
        ... 10 more
Caused by: jakarta.enterprise.inject.spi.DeploymentException: Method 'get' of class 'org.acme.SomePage' is considered a non blocking method. @RunOnVirtualThread can only be used on  methods considered blocking
        at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.isRunOnVirtualThread(EndpointIndexer.java:853)
        at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createResourceMethod(EndpointIndexer.java:729)
        ... 12 more
```

### Expected behavior

The application builds, and invokation is done on virtual thread.

### Actual behavior

Application does not build

### How to Reproduce?

Use the app generator, generate a app with quarkus-resteasy-reactive-qute. Add @RunOnVirtualThread in `SomePage` on class or method level. 

### Output of `uname -a` or `ver`

linux

### Output of `java -version`

21

### Quarkus version or git rev

2.4.2

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.3

### Additional information

![Screenshot from 2023-10-20 10-45-19](https://github.com/quarkusio/quarkus/assets/769579/8f2d6aa9-72a6-4473-b1ad-db7331b067aa)

It at least looks like I was in a confused state when creating the sample. The original exception is not the correct one. 
I have updated it now. 
The core exception is 
```
Caused by: jakarta.enterprise.inject.spi.DeploymentException: Method 'get' of class 'org.acme.SomePage' is considered a non blocking method. @RunOnVirtualThread can only be used on  methods considered blocking
``

--- a/docs/src/main/asciidoc/qute-reference.adoc
+++ b/docs/src/main/asciidoc/qute-reference.adoc
@@ -2553,7 +2553,9 @@ If instead you are using RESTEasy Reactive via the `quarkus-resteasy-reactive` e
 </dependency>
 ----
 
-Both of these extensions register a special `ContainerResponseFilter` implementation which enables resource methods to return a `TemplateInstance`, thus freeing users of having to take care of all necessary internal steps.
+Both of these extensions register a special response filter which enables resource methods to return a `TemplateInstance`, thus freeing users of having to take care of all necessary internal steps.
+
+NOTE: If using RESTEasy Reactive, a resource method that returns `TemplateInstance` is considered non-blocking. You need to annotate the method with `io.smallrye.common.annotation.Blocking` in order to mark the method as blocking. For example if it's also annotated with `@RunOnVirtualThread`.
 
 The end result is that a using Qute within a Jakarta REST resource may look as simple as:
 

